codex/contracts/market.nim

```diff
@@ -502,3 +527,35 @@ method queryPastStorageRequestedEvents*(
     let fromBlock = await market.contract.provider.pastBlockTag(blocksAgo)
 
     return await market.queryPastStorageRequestedEvents(fromBlock)
+
+method slotCollateral*(
+    market: OnChainMarket, requestId: RequestId, slotIndex: UInt256
+): Future[?UInt256] {.async: (raises: []).} =
+  let slotid = slotId(requestId, slotIndex)
+
+  try:
+    let slotState = await market.slotState(slotid)
+
+    without request =? await market.getRequest(requestId):
+      return UInt256.none
+
+    return market.slotCollateral(request.ask.collateralPerSlot, slotState)
+  except CatchableError as err:
```

Comment:
```suggestion
  except MarketError as err:
```
This change will only work if the market abstraction changes i suggested above were made. Doing this will also not swallow CancelledErrors, stopping cancellation propagation.